Core/src/Vertexing/KalmanVertexUpdater.cpp

```diff
@@ -206,19 +227,20 @@ void KalmanVertexUpdater::detail::calculateUpdate4(
   calculateUpdateImpl<4>(vtxPos, vtxCov, linTrack, trackWeight, sign, cache);
 }
 
-void KalmanVertexUpdater::detail::updateVertexWithTrack(
-    Vector4& vtxPos, SquareMatrix4& vtxCov,
-    std::pair<double, double>& fitQuality, TrackAtVertexRef trk, int sign,
-    unsigned int nDimVertex) {
+void KalmanVertexUpdater::updateVertexWithTrack(Vertex& vtx, TrackAtVertex& trk,
+                                                unsigned int nDimVertex) {
+  std::pair<double, double> fitQuality = vtx.fitQuality();
+
   if (nDimVertex == 3) {
-    update<3>(vtxPos, vtxCov, fitQuality, trk, sign);
+    update<3>(vtx.fullPosition(), vtx.fullCovariance(), fitQuality, trk, 1);
   } else if (nDimVertex == 4) {
-    update<4>(vtxPos, vtxCov, fitQuality, trk, sign);
+    update<4>(vtx.fullPosition(), vtx.fullCovariance(), fitQuality, trk, 1);
   } else {
     throw std::invalid_argument(
         "The vertex dimension must either be 3 (when fitting the spatial "
         "coordinates) or 4 (when fitting the spatial coordinates + time).");
   }
+  vtx.setFitQuality(fitQuality);
```

Comment:
was this missing before or did it happen in another function?

Reply:
It was happening inside the update function. I pulled this out originally I think to work around the `Vertex` being templated, but it's not anymore. I can revert this back to accepting the vertex itself, and then the fitQuality dance goes away.